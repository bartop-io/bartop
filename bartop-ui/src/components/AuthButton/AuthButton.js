import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { actions } from '../../ducks/authentication/authentication';
import strings from '../../strings';

const Button = styled.button`
  border: none;
`;

export const AuthButton = ({ loggedIn, login, logout }) =>
  !loggedIn ? (
    <Button onClick={login}>{strings.auth.login}</Button>
  ) : (
    <Button onClick={logout}>{strings.auth.logout}</Button>
  );

AuthButton.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  loggedIn: state.authentication.status.loggedIn
});

const mapDispatchToProps = {
  login: actions.startLogin,
  logout: actions.logout
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthButton);
