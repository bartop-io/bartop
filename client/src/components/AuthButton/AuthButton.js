import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { actions } from '../../ducks/authentication/authentication';

const Button = styled.button`
  border: none;
`;

export const AuthButton = ({ loggedIn, login, logout }) =>
  !loggedIn ? (
    <Button onClick={login}>Login</Button>
  ) : (
    <Button onClick={logout}>Logout</Button>
  );

AuthButton.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  login: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired
};

AuthButton.defaultProps = {
  expiresAt: null
};

const mapStateToProps = state => ({
  loggedIn: state.authentication.status.loggedIn
});

const mapDispatchToProps = {
  login: actions.loginRequest,
  logout: actions.logout
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthButton);
