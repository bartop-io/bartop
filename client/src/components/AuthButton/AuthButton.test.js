import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { AuthButton } from './AuthButton';
import { noop } from '../../testUtils';

const requiredProps = {
  loggedIn: false,
  login: noop,
  logout: noop
};

it('matches the snapshot', () => {
  const wrapper = shallow(<AuthButton {...requiredProps} />);
  expect(wrapper).toMatchSnapshot();
});

it('if logged out, shows login text', () => {
  const wrapper = shallow(<AuthButton {...requiredProps} />);
  expect(wrapper.dive().text()).toEqual('Login');
});

it('if logged out, calls login if clicked', () => {
  const login = jest.spyOn(requiredProps, 'login');
  const wrapper = shallow(<AuthButton {...requiredProps} />);
  expect(login).not.toBeCalled();
  wrapper.dive().simulate('click');
  expect(login).toBeCalled();
  login.mockRestore();
});

it('if logged in, shows logout text', () => {
  const wrapper = shallow(<AuthButton {...requiredProps} loggedIn />);
  expect(wrapper.dive().text()).toEqual('Logout');
});

it('if logged in, calls logout if clicked', () => {
  const logout = jest.spyOn(requiredProps, 'logout');
  const wrapper = shallow(<AuthButton {...requiredProps} loggedIn />);
  expect(logout).not.toBeCalled();
  wrapper.dive().simulate('click');
  expect(logout).toBeCalled();
  logout.mockRestore();
});
