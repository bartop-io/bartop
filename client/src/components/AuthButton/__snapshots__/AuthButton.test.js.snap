// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<styled.button
  onClick={[Function]}
>
  Login
</styled.button>
`;
